### DISCUSSION TITLE
Is S3-compliant storage supported or only specifically AWS S3?

### DISCUSSION THREAD

**Discussion by danielrosehill** [outline/outline#7834] [Self Hosting]
I see that in the template `.docker.env` there's this comment for the `FILE_STORAGE` parameter:

>  Specify what storage system to use. Possible value is one of "s3" or "local".  
> For "local", the avatar images and document attachments will be saved on local disk.    
> FILE_STORAGE=local  

Later in the file there are some more comments:

> To support uploading of images for avatars and document attachments in a distributed 
> architecture an s3-compatible storage can be configured if FILE_STORAGE=s3 above.
> AWS_ACCESS_KEY_ID=get_a_key_from_aws
> AWS_SECRET_ACCESS_KEY=get_the_secret_of_above_key
> AWS_REGION=xx-xxxx-x
> AWS_S3_ACCELERATE_URL=
> AWS_S3_UPLOAD_BUCKET_URL=http://s3:4569
> AWS_S3_UPLOAD_BUCKET_NAME=bucket_name_here
> AWS_S3_FORCE_PATH_STYLE=true
> AWS_S3_ACL=private

From the way the first comment is phrased, it sounds like only AWS S3 can be configured to hold file storage, but from the second it sounds like S3-compliant storage would work (but then the environment variables are AWS-specific).

So, I'm confused!

Is it possible to use an S3-compliant storage like Minio or B2 or does it specifically have to be AWS S3?

**Comment by tommoor** [comment]
It is possible to use a service that is API-compatible, a list of services that have been used can be found here:

https://docs.getoutline.com/s/hosting/doc/file-storage-N4M0T6Ypu7#h-s3-compatible-services
